File: scilab/modules/core/sci_gateway/fortran/sci_getcwd.f

      subroutine intsgetdir(fname)
c     --------------------------
      character*(*) fname
      logical checkrhs,checklhs
      include 'stack.h'
      logical putlhsvar, createvarfromptr
      double precision l1
      nbvars = 0
      rhs = max(0,rhs)
      if(.not.checkrhs(fname,0,0)) return
      if(.not.checklhs(fname,1,1)) return
      call scigetcwd(l1,m,err)
      if(err .gt. 0) then 
         buf = fname // ': Internal Error' 
         call error(998)
         return
      endif
      if(.not.createvarfromptr(1,'c',m,1,l1)) return
      lhsvar(1)=1
      if(.not.putlhsvar()) return
      end

File: scilab/modules/core/sci_gateway/c/sci_getcwd.c
/*-----------------------------------------------------------------------------------*/
/* INRIA 2006 */
/* Allan CORNET */
/*-----------------------------------------------------------------------------------*/ 
#include "gw_core.h"
#include "machine.h"
#include "stack-c.h"
#include "scicurdir.h"
#include "MALLOC.h"
#include "error_scilab.h"
/*-----------------------------------------------------------------------------------*/
int C2F(sci_getcwd) _PARAMS((char *fname,unsigned long fname_len))
{
	int ierr=0;
	int lpath=0;
	char *path=NULL;

	Rhs=Max(Rhs,0);

	CheckRhs(0,0);
	CheckLhs(0,1);

	C2F(scigetcwd)(&path,&lpath,&ierr);
	
	if (ierr)
	{
		error_scilab(998,_("Internal Error."));
		return 0;
	}
	else
	{
		static int n1,m1;
		n1=1;
		CreateVarFromPtr(Rhs+ 1,STRING_DATATYPE,(m1=(int)strlen(path), &m1),&n1,&path);
		LhsVar(1) = Rhs+1;
		C2F(putlhsvar)();	
	}

	return 0;
}
/*-----------------------------------------------------------------------------------*/
